### PR TITLE
Allow any authenticated user to POST to /fcr:tx

### PR DESCRIPTION
**Pull request by peichman-umd**
**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3321

# What does this Pull Request do?

Allows any authenticated user to create a transaction by allowing POST to the /fcr:tx endpoint as long as the current user is authenticated.

# How should this be tested?

* Run the WebACRecipesIT integration tests
* Build and run using `mvn jetty:run` then test by doing: `curl -u testuser:testpass -i -X POST http://localhost:8080/rest/fcr:tx`. You should get a `201 Created` response.

# Interested parties
@fcrepo4/committers, @awoods

### CODE REVIEW

File: fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/FedoraTypes.java [hunk 1]
```diff
@@ -99,6 +99,8 @@
 
     String FCR_TOMBSTONE = "fcr:tombstone";
 
+    String FCR_TX = "fcr:tx";
```
Comment (awoods) [inline fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/FedoraTypes.java:102]: This is already defined in [TransactionConstants](https://github.com/fcrepo4/fcrepo4/blob/main/fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/session/TransactionConstants.java#L42)... although FedoraTypes seems like a logical home. Would you be interested in refactoring these into a single constant?

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACFilter.java [hunk 2]
```diff
@@ -365,6 +367,10 @@ private boolean isAuthorized(final Subject currentUser, final HttpServletRequest
                 }
             }
         case "POST":
+            if (isTxEndpoint && currentUser.isAuthenticated()) {
+                log.debug("POST allowed to transaction endpoint");
```
Comment (awoods) [inline fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACFilter.java:371]: Could you add the "currentUser"s ID to the debug message?

**Review by awoods — CHANGES_REQUESTED** [review]
This works as desired. Nice job.

File: fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/FedoraTypes.java (hunk 1 above)
Comment (peichman-umd) [reply fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/FedoraTypes.java:102]: @awoods I've modified `TX_PREFIX` to be defined in terms of the `FCR_TX` constant.

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACFilter.java (hunk 2 above)
Comment (peichman-umd) [reply fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACFilter.java]: Done.

**Review by awoods — APPROVED** [review]

**Pull request merged**